plugins/woocommerce/src/Blocks/BlockTemplatesRegistry.php

```diff
@@ -0,0 +1,44 @@
+<?php
+namespace Automattic\WooCommerce\Blocks;
+
+use Automattic\WooCommerce\Blocks\Templates\AbstractTemplate;
+use Automattic\WooCommerce\Blocks\Templates\AbstractTemplatePart;
+
+/**
+ * BlockTemplatesRegistry class.
+ *
+ * @internal
+ */
+class BlockTemplatesRegistry {
+
+	/**
+	 * The array of registered templates.
+	 *
+	 * @var AbstractTemplate[]|AbstractTemplatePart[]
+	 */
+	private static $templates = array();
```

Comment:
We cache the instance of this class in the container so this property doesn't have to be static right?

Reply:
This has also changed in 48800e5c55ee2d74f2515eb502cfce8d6dfc9e3e.

Reply:
The main reason I left `$templates` and the other methods in this class static is because I don't expect them to be dependent on a specific instance of the class. In other words, I wouldn't expect more than one `BlockTemplatesRegistry` to exist in the app. Keeping the variable and the methods static guarantees that there will never be two instances of `BlockTemplatesRegistry` with different `$templates` registered.

Having said that, I don't have a strong opinion and I'm happy to change that. It's just that I'm not sure to understand what are the benefits of making them non-static. I'm not a PHP expert so sorry if I'm missing something obvious! :sweat_smile: